pkgs/by-name/en/encore/tsparser.nix

```diff
@@ -0,0 +1,72 @@
+{
+  runtimesPath,
+  rustPlatform,
+  makeWrapper,
+  pkg-config,
+  protobuf,
+  openssl,
+  version,
+  src,
+  lib,
+  ...
+}: let
+  cargoLockFile = ./tsparser-Cargo.lock;
+in
+  rustPlatform.buildRustPackage {
+    pname = "tsparser-encore";
+
+    inherit src version;
+
+    nativeBuildInputs = [
+      makeWrapper
+      pkg-config
+      openssl
+    ];
+
+    buildInputs = [
+      protobuf
+    ];
+
+    buildAndTestSubdir = "tsparser";
+
+    doCheck = false;
+
+    patchPhase = ''
+      runHook prePatch
+
+      # the build complains about the diff between both cargo.lock files and
+      # is necessary to have a cargo.lock since it contains git deps
+      rm -rf tsparser/Cargo.lock
+      cp ${cargoLockFile} Cargo.lock
+
+      # to avoid to compile "broken" members
+      rm -rf runtimes
+      substituteInPlace Cargo.toml \
+        --replace 'members = ["runtimes/core", "runtimes/js", "tsparser"]' 'members = ["tsparser"]' \
+
+      substituteInPlace tsparser/src/bin/tsparser-encore.rs \
```

Comment:
it is strange that you need to do these substitutions :thinking: what rust version is this built with?

Reply:
(we always build with latest stable, so currently 1.79.0)

Reply:
In the unstable channel we are at [1.78.0](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=rustc)

![image](https://github.com/user-attachments/assets/85e1fcf1-08a4-4651-9393-f6b6826c39df)

Can you test that version by changing it in your local toolchain?

Reply:
I tried downgrading, and I got the same issue on 1.78.0. Would it be an option to use something like fenix? they have a more up to date rust version.